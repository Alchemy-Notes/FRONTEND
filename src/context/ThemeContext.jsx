import { createContext, useContext, useState } from 'react';

const ThemeContext = createContext();

const ThemeProvider = ({ children }) => {
  const [theme, setTheme] = useState(false);

  return (
    <ThemeContext.Provider value={{ theme, setTheme }}>
      {children}
    </ThemeContext.Provider>
  );
};

const useTheme = () => {
  const context = useContext(ThemeContext);

  if (context === undefined) {
    throw new Error('useTheme must be used within a ThemeContext Provider');
  }

  return context;
};

export { ThemeProvider, useTheme };
