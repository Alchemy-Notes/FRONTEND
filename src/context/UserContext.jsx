import { createContext, useContext, useEffect, useState } from 'react';
import { getMe } from '../services/auth';
import { getUserNotes } from '../services/notes';

const UserContext = createContext();

const UserProvider = ({ children }) => {
  const [user, setUser] = useState(null);
  const [notes, setNotes] = useState(null);
  const [tree, setTree] = useState(null);

  useEffect(() => {
    async function get() {
      const result = await getMe();
      setUser(result);
    }
    get();
  }, []);

  useEffect(() => {
    async function getNotes() {
      const result = await getUserNotes({
        userId: user.id,
        query: { type: 'favorite' },
      });
      setNotes(result);
    }
    if (user && user.id) getNotes();
  }, [user]);

  return user && notes ? (
    <UserContext.Provider
      value={{ user, setUser, notes, setNotes, tree, setTree }}
    >
      {children}
    </UserContext.Provider>
  ) : (
    <></>
  );
};

const useUser = () => {
  const context = useContext(UserContext);

  if (context === undefined) {
    throw new Error('useUser must be used within a UserContext Provider');
  }

  return context;
};

export { UserProvider, useUser };
