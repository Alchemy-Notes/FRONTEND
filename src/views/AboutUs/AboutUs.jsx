import linkdinbtn from '../../../public/assets/linkdinbtn.jpeg';
import GitHub_Logo from '../../../public/assets/GitHub_Logo.png';
import AliceTwitterButtonLong from '../../../public/assets/AliceTwitterButtonLong.jpeg';

export default function AboutUs() {
  return (
    // className={styles.aboutus}
    <>
      <section className="bio">
        <h4>Triana Cerda</h4>
        <div>
          {/* className={styles.linkdin} */}
          <a href="http://www.linkedin.com/in/triana-cerda/">
            <img src={linkdinbtn} alt="Visit my LinkedIn Profile" />
          </a>

          {/* className={styles.github-button} */}
          <a
            href="https://github.com/trianacerda"
            data-color-scheme="no-preference: dark; light: dark; dark: dark;"
            aria-label="Follow @trianacerda on GitHub"
          >
            <img src={GitHub_Logo} alt="github" />
          </a>
        </div>
      </section>

      <section className="bio">
        <h4>Pete Hamrick</h4>
        <div>
          {/* className={styles.linkdin} */}
          <a href="https://www.linkedin.com/in/petehamrick/">
            <img src={linkdinbtn} alt="Visit my LinkedIn Profile" />
          </a>

          {/* className={styles.github-button} */}
          <a
            href="https://github.com/pete-hamrick"
            data-color-scheme="no-preference: dark; light: dark; dark: dark;"
            aria-label="Follow @trianacerda on GitHub"
          >
            <img src={GitHub_Logo} alt="github" />
          </a>
        </div>
      </section>

      <section className="bio">
        <h4>Alice Retz</h4>
        <div>
          <a className="linkdin" href="http://www.linkedin.com/in/aliceretz/">
            <img src={linkdinbtn} alt="Visit my LinkedIn Profile" />
          </a>

          <a
            className="github-button"
            href="https://github.com/Alice-Retz"
            data-color-scheme="no-preference: dark; light: dark; dark: dark;"
            aria-label="Follow @Alice Retz(she/they) on GitHub"
          >
            <img src={GitHub_Logo} alt="github" />
          </a>

          <a className="twitter-button" href="https://twitter.com/AliceRetz">
            <img src={AliceTwitterButtonLong} alt="twitter" />
          </a>
        </div>
      </section>

      <section className="bio">
        <h4>Peaul Brubaker</h4>
        <div>
          {/* className={styles.linkdin} */}
          <a href="https://www.linkedin.com/in/paul-brubaker-a94597176/">
            <img src={linkdinbtn} alt="Visit my LinkedIn Profile" />
          </a>

          {/* className={styles.github-button} */}
          <a
            href="https://github.com/p-brubaker"
            data-color-scheme="no-preference: dark; light: dark; dark: dark;"
            aria-label="Follow @trianacerda on GitHub"
          >
            <img src={GitHub_Logo} alt="github" />
          </a>
        </div>
      </section>
    </>
  );
}
