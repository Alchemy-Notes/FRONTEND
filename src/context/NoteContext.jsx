import { createContext, useContext, useEffect, useState } from 'react';
import { getUserNotes } from '../services/notes';
import { useUser } from './UserContext';

const NoteContext = createContext();

const NoteProvider = ({ children }) => {
  const [notes, setNotes] = useState([]);
  const { user } = useUser();

  useEffect(() => {
    async function get() {
      const result = await getUserNotes({
        userId: user.id,
        query: { type: 'favorite' },
      });
      setNotes(result);
    }
    if (user && user.id) get();
  }, []);

  return (
    <NoteContext.Provider value={{ notes, setNotes }}>
      {children}
    </NoteContext.Provider>
  );
};

const useNotes = () => {
  const context = useContext(NoteContext);

  if (context === undefined) {
    throw new Error('useNotes must be used within a NoteContext Provider');
  }

  return context;
};

export { NoteProvider, useNotes };
