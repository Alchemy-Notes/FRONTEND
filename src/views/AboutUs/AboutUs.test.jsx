import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { ThemeProvider } from '../../context/ThemeContext';
import AboutUs from './AboutUs';

it('should render AboutUs comp', () => {
  const { container } = render(
    <MemoryRouter>
      <ThemeProvider>
        <AboutUs />
      </ThemeProvider>
    </MemoryRouter>
  );

  expect(container).toMatchSnapshot();
});
