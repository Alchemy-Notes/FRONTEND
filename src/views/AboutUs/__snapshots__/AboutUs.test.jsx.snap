// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AboutUs comp 1`] = `
<div>
  <div
    class="wrapper"
  >
    <h1>
      Dev Notes
    </h1>
    <div
      class="aboutComp"
    >
      <section
        class="lightAbout"
      >
        <h2>
          triana(Cerda)
        </h2>
        <div
          class="lightAboutLinks"
        >
          contact(Me) =&gt; {
          <p
            style="margin-left: 20px;"
          >
            <a
              href="http://www.linkedin.com/in/triana-cerda/"
            >
              <img
                alt="Visit my LinkedIn Profile"
                src="/path/image"
              />
            </a>
            ;
          </p>
          <p
            style="margin-left: 20px;"
          >
            <a
              aria-label="Follow @trianacerda on GitHub"
              data-color-scheme="no-preference: dark; light: dark; dark: dark;"
              href="https://github.com/trianacerda"
            >
              <img
                alt="github"
                class="lightGithubLogoImg"
                src="/path/image"
              />
            </a>
            ;
          </p>
          }
        </div>
      </section>
      <section
        class="lightAbout"
      >
        <h2>
          Pete.Hamrick
        </h2>
        <div
          class="lightAboutLinks"
        >
          &lt;h1&gt; Contact Me &lt;/h1&gt;
          <br />
          &lt;div&gt;
          <p
            style="margin-left: 20px;"
          >
            <a
              href="https://www.linkedin.com/in/petehamrick/"
            >
              <img
                alt="Visit my LinkedIn Profile"
                src="/path/image"
              />
            </a>
          </p>
          <p
            style="margin-left: 20px;"
          >
            &lt;p /&gt;
          </p>
          <p
            style="margin-left: 20px;"
          >
            <a
              aria-label="Follow @pete-hamrick on GitHub"
              data-color-scheme="no-preference: dark; light: dark; dark: dark;"
              href="https://github.com/pete-hamrick"
            >
              <img
                alt="github"
                class="lightGithubLogoImg"
                src="/path/image"
              />
            </a>
          </p>
          <br />
          &lt;/div&gt;
        </div>
      </section>
    </div>
    <div
      class="aboutComp"
    >
      <section
        class="lightAbout"
      >
        <h2>
          Alice:Retz
        </h2>
        <div
          class="lightAboutLinks"
        >
          .contactMe {
          <p
            style="margin-left: 20px;"
          >
            <a
              class="linkdin"
              href="http://www.linkedin.com/in/aliceretz/"
            >
              <img
                alt="Visit my LinkedIn Profile"
                src="/path/image"
              />
            </a>
          </p>
          <p
            style="margin-left: 20px;"
          >
            <a
              aria-label="Follow @Alice-Retz on GitHub"
              class="github-button"
              data-color-scheme="no-preference: dark; light: dark; dark: dark;"
              href="https://github.com/Alice-Retz"
            >
              <img
                alt="github"
                class="lightGithubLogoImg"
                src="/path/image"
              />
            </a>
          </p>
          <p
            style="margin-left: 20px;"
          >
            <a
              class="twitter-button"
              href="https://twitter.com/AliceRetz"
            >
              <img
                alt="twitter"
                src="/path/image"
              />
            </a>
          </p>
          <br />
           }
        </div>
      </section>
      <section
        class="lightAbout"
      >
        <h2>
          **Paul Brubaker**
        </h2>
        <div
          class="lightAboutLinks"
        >
          <br />
          try {
          <p
            style="margin-left: 20px;"
          >
            if (contactMe) {
          </p>
          <p
            style="margin-left: 40px;"
          >
            <a
              class="linkdin"
              href="https://www.linkedin.com/in/paul-brubaker-a94597176/"
            >
              <img
                alt="Visit my LinkedIn Profile"
                src="/path/image"
              />
            </a>
          </p>
          <p
            style="margin-left: 20px;"
          >
            } else {
          </p>
          <p
            style="margin-left: 40px;"
          >
            <a
              aria-label="Follow @p-brubaker on GitHub"
              class="github-button"
              data-color-scheme="no-preference: dark; light: dark; dark: dark;"
              href="https://github.com/p-brubaker"
            >
              <img
                alt="github"
                class="lightGithubLogoImg"
                src="/path/image"
              />
            </a>
          </p>
          <p
            style="margin-left: 20px;"
          >
            }
          </p>
          <p>
            }
          </p>
        </div>
      </section>
    </div>
  </div>
</div>
`;
